tools/proto-compiler/src/constants.rs

```diff
@@ -85,14 +86,22 @@ pub static CUSTOM_FIELD_ATTRIBUTES: &[(&str, &str)] = &[
         ".tendermint.types.EvidenceParams.max_bytes",
         QUOTED_WITH_DEFAULT,
     ),
-    (".tendermint.abci.ResponseInfo.last_block_height", QUOTED),
     (".tendermint.version.Consensus.block", QUOTED),
     (".tendermint.version.Consensus.app", QUOTED_WITH_DEFAULT),
+    (".tendermint.abci.ResponseInfo.data", DEFAULT),
+    (".tendermint.abci.ResponseInfo.version", DEFAULT),
+    (
+        ".tendermint.abci.ResponseInfo.app_version",
+        QUOTED_WITH_DEFAULT,
+    ),
+    (
+        ".tendermint.abci.ResponseInfo.last_block_height",
+        QUOTED_WITH_DEFAULT,
+    ),
```

Comment:
Not sure if it's best to fix it here with compiler tweaks, or maybe move the serde annotations to the `AbciInfo` struct, where the serde traits are really used, while currently being delegated to `tendermint_proto::abci::ResponseInfo`.
We have to customize serde for every field of the involved struct in either case, so doing it directly in source would be cleaner.